### PR TITLE
transform to python2 version

### PR DESCRIPTION
**Pull request by hyzhan**
Thank you very much for your code. This helped me a lot.
I had tested these scirpt on word model:
python demo_server.py --checkpoint tacotron-20170720/model.ckpt
python preprocess.py --dataset ljspeech
python train.py (10 step only)
python eval.py --checkpoint --checkpoint tacotron-20170720/model.ckpt
And change util/cmudict.py to python2 version.
I hope it can be helpful to you. 

If there is any problem, please feel free to contact me. My email is hyzhan1993@hotmail.com.

### CODE REVIEW

**Comment by DarkDefender** [comment]
Will this still work with python3? I'm asking because there are quite a few linux distributions that have or will soon make `python` be python3.

If it works with both, great! But it it now only works with python2 I would consider this a step back in functionality

File: datasets/ljspeech.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def build_from_path(in_dir, out_dir, num_workers=1, tqdm=lambda x: x):
   executor = ProcessPoolExecutor(max_workers=num_workers)
   futures = []
   index = 1
-  with open(os.path.join(in_dir, 'metadata.csv'), encoding="utf-8") as f:
+  with open(os.path.join(in_dir, 'metadata.csv'), 'rb') as f:
```
Comment (keithito) [inline datasets/ljspeech.py]: I think the recommended way of doing this in Python 2 is to use io.open. Then you can continue to pass an encoding.
https://mail.python.org/pipermail/python-list/2015-March/687124.html

File: preprocess.py [hunk 2]
```diff
@@ -15,15 +15,17 @@ def preprocess_blizzard(args):
 
 
 def preprocess_ljspeech(args):
-  in_dir = os.path.join(args.base_dir, 'LJSpeech-1.0')
-  out_dir = os.path.join(args.base_dir, args.output)
-  os.makedirs(out_dir, exist_ok=True)
+  in_dir = os.path.join(os.getcwd(), 'LJSpeech-1.0')
```
Comment (keithito) [inline preprocess.py]: Why did this change from args.base_dir to os.getcwd?

File: preprocess.py [hunk 3]
```diff
@@ -15,15 +15,17 @@ def preprocess_blizzard(args):
...
+  if not os.path.exists(out_dir):
+      os.makedirs(out_dir)
+  # os.makedirs(out_dir, exist_ok=True)
   metadata = ljspeech.build_from_path(in_dir, out_dir, args.num_workers, tqdm=tqdm)
   write_metadata(metadata, out_dir)
 
 
 def write_metadata(metadata, out_dir):
-  with open(os.path.join(out_dir, 'train.txt'), 'w', encoding="utf-8") as f:
+  with open(os.path.join(out_dir, 'train.txt'), 'wb') as f:
```
Comment (keithito) [inline preprocess.py]: See other comment about io.open.

File: train.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
 from models import create_model
 from util import audio, infolog, plot, textinput, ValueWindow
 log = infolog.log
+os.environ["CUDA_VISIBLE_DEVICES"]="1"
```
Comment (keithito) [inline train.py]: This change seems unrelated to python 2 support.

**Comment by feiyuhong** [comment]
Waves are all silent when I use python2, do you know why?

File: datasets/ljspeech.py (hunk 1 above)
Comment (StanDex) [reply datasets/ljspeech.py]: I have a problem with launching your system with your trained model checkpoint. Please help me! I described my problem here: https://github.com/keithito/tacotron/issues/94 Thanks
